docs/ocaml/00-outline.md

```diff
@@ -0,0 +1,143 @@
+1. Intro
+  - Why Functional Programming?
+    - mutability
+    - encapsulation
+    - functional programming in Java is tedious (but possible)
```

Comment:
maybe mention about python and their use of functions as first-class values. I was using the decorator functionality in Python and I thought it was ridiculous.